aten/src/ATen/core/TensorImpl.h

```diff
@@ -1452,6 +1380,80 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
 
 };
 
+/**
+ * @brief Copies the data from a source tensor, with a contex provided to
+ * carry out the underlying memcpy operation.  This method respects
+ * caffe2_keep_on_shrink.
+ *
+ * After CopyFrom, this function guarantees that the destination tensor will
+ * have the same initialization state and dtype as src.  This function
+ * preserves the DeviceType of the source tensor (so, e.g., if you allocate
+ * a tensor on CPU and then CopyFrom a CUDA tensor, that will to a
+ * CUDA-to-CPU transfer).
+ *
+ * 'async' parameter triggers async copy for CUDA tensors
+ */
+inline void CopyToDevice(TensorImpl* dst, const TensorImpl& src, bool async = false) {
```

Comment:
actually, maybe move this function to Caffe2's Tensor class then? Rationale - we probably don't want  it to be in merged tensor public API, but it might stick around for a while in existing code

Reply:
ok, moved it to a caffe2::Tensor method instead.